src/cmd/run.js

```diff
@@ -62,24 +65,36 @@ export function defaultWatcherCreator(
   return onSourceChange({
     sourceDir,
     artifactsDir,
-    onChange: () => {
-      log.debug(`Reloading add-on ID ${addonId}`);
-
-      return client.reloadAddon(addonId)
-        .catch((error) => {
-          log.error('\n');
-          log.error(error.stack);
-          desktopNotifications({
-            title: 'web-ext run: error occurred',
-            message: error.message,
-          });
-          throw error;
-        });
-    },
+    onChange: () => addonReload({addonId, client}),
```

Comment:
If I'm not wrong, these `addonReload/ ReloadParams/ defaultAddonReload` can be reverted: in the current version of this change, the `addonReload` helper seems to be no different from the previous version of this `onChange` callback.

Reverting these changes should help to greatly simplify this change.